src/uast.h

```diff
@@ -0,0 +1,75 @@
+#ifndef _NODE_API
+#define _NODE_API
+
+#include <libxml/parser.h>
+#include <libxml/tree.h>
+#include <libxml/xpath.h>
+#include <libxml/xpathInternals.h>
+
+#include "node-iface.h"
+
+typedef struct _find_ctx {
```

Comment:
It'd probably be better to make this struct private and offer a public api instead, so that it can be changed in the future without compatibility problems, but as you prefer